appinventor/components/src/com/google/appinventor/components/runtime/util/TextViewUtil.java

```diff
@@ -168,13 +168,11 @@ public static Typeface getTypeFace(Form form, String fontFile) {
     Typeface typeface;
     if (!fontFile.contains("/")) {
       if (form instanceof ReplForm) {
-        if (Build.VERSION.SDK_INT > 28) {
-          File file = new File("/storage/emulated/0/Android/data/edu.mit.appinventor.aicompanion3/files/assets/" + fontFile);
-          typeface = Typeface.createFromFile(file);
-        } else {
-          File file = new File("/storage/emulated/0/AppInventor/assets/" + fontFile);
-          typeface = Typeface.createFromFile(file);
-        }
+        try {
+          File file = new File(MediaUtil.fileUrlToFilePath(form.getAssetPath(fontFile)));
+        } catch (IllegalArgumentException e) {
+        } catch (Exception e) {}
```

Comment:
This should be catching IOException here rather than IllegalArgumentException and Exception generally. It would also be good if we report an error via the form so that the developer knows there is a problem loading the font.

Reply:
Please answer the questions so I can continue.